canonical/tempo-k8s-operator
# Clean up tracing databag in v2

## Issue
Fixes #105

## Solution
- update provider app databag to follow https://github.com/canonical/charm-relation-interfaces/pull/150
- update minor version for tracing library (2.6)
- removed tracing library v1 support
- fix UTs, Scenario tests, Integration tests


## Testing instructions

with tempo charm repo root as CWD:
```
juju add-model tempo-clite-test
juju deploy cos-lite --channel edge --trust

git fetch
git checkout OPENG-2399
jhack sync "*" --touch ./lib/charms/tempo_k8s/v2/tracing.py

charmcraft pack
jhack deploy 

jhack imatrix fill
```

now go to the grafana dashboard and ensure you can see charm traces for all apps (and grafana workload)

add tls:
```
juju deploy self-signed-certificates ssc
jhack imatrix fill
```

verify you can still see the new charm traces for all apps (and grafana workload)


Issues:
traefik gets stuck because we removed the deprecated api that traefik was still relying on. see https://github.com/canonical/traefik-k8s-operator/pull/364

Same for grafana: https://github.com/canonical/grafana-k8s-operator/pull/333

Catalogue has broken tls charm tracing
https://github.com/canonical/catalogue-k8s-operator/pull/102
